src/client/dfs/SConscript

```diff
@@ -54,19 +56,45 @@ def scons():
                         instver = cache["lustre-client"].installed
             if gotpkg is True:
                 if parse_version(instver.version) >= parse_version('2.12.57'):
+                    """ Lustre version is compatible """
                     gotversion = True
         if gotdistro is False:
             print("Unable to identify distro.")
+            raise ImportError('fallback')
         else:
             if gotpkg is False:
                 print("Unable to find lustre nor lustre-client package.")
+                raise ImportError('fallback')
             else:
                 if gotversion is False:
-                    print("Unable to find minimum version of lustre nor lustre-client package.")
+                    print("Installed Lustre version is not compatible.")
                 else:
-                    denv.AppendUnique(CCFLAGS=['-DLUSTRE_INCLUDE'])
-    except ImportError:
-        raise
+                    print("Installed Lustre version is compatible.")
+    except ImportError as error:
+        print(error.message +
+              ", trying to find Lustre headers in usual locations.")
+        """ Try the empirical way, by just trying to check both required header files are present ... """
+        if path.exists("/usr/include/lustre/lustreapi.h"):
+            if path.exists("/usr/include/linux/lustre/lustre_user.h"):
+                gotpkg = True
+                with open('/usr/include/linux/lustre/lustre_user.h', 'r') as file:
+                    for line in file:
+                        """ if definition present Lustre version is compatible """
+                        if 'LU_FOREIGN_TYPE_SYMLINK' in line:
```

Comment:
This is a typical example of an API feature test that you can have `scons` do.  This code should be using that, not writing it's own python to guesstimate.

Reply:
Well, this is where my "scons" knowledge finds it is very low limit... BTW, do you know what a scons is in french, but may be you also use this name?

Reply:
But Scons is the tool of choice for this project.  For maintenance (in the least) we should be using the features of of the tool of choice and not end up rewriting it in free-flowing Python code that will increase the maintenance burden.